### PR TITLE
Use RandomSearchNextPoints for random search.

### PR DESCRIPTION
**Pull request by harvey-intel**
Using the views method instead of creating the `CategoricalDomain` for generating random points.

Will merge after `rs-view` branch is merged into `libsigopt` main.
Temporarily change the reqs in here to get the unit tests working .Will revert to point to `libsigopt` main again before mergin.

### CODE REVIEW

File: src/python/zigopt/sigoptcompute/adapter.py [hunk 1]
```diff
@@ -237,6 +238,21 @@ def multisolution_best_assignments(
...
+  def random_search_next_points(
+    self,
+    experiment,
+    num_to_suggest,
+    tag=None,
+  ):
+    view_input = {
+      "domain_info": self.generate_domain_info(experiment, only_active_categorical_values=True),
+      "num_to_sample": num_to_suggest,
+      "tag": self.supplement_tag_with_experiment_id(tag, experiment),
```
Comment (erichanslee) [inline src/python/zigopt/sigoptcompute/adapter.py:250]: If we aren't using the tag might as well pass in an empty dict

File: requirements.txt [hunk 2]
```diff
@@ -6,7 +6,7 @@ bcrypt==3.2.0
 boto3==1.26.99
 botocore==1.29.99
 cffi==1.15.1
-click==8.0.4
+click==8.1.3
```
Comment (erichanslee) [inline requirements.txt:9]: I assume this is here to pass some failing tests?

**Review by erichanslee — APPROVED** [review]
Thanks! `generate_random_suggestion_datas` is cleaner now.

**Comment by malkomes** [comment]
I suppose we have to change the `from libsigopt.compute` parts too

File: src/python/zigopt/sigoptcompute/adapter.py (hunk 1 above)
Comment (malkomes) [reply src/python/zigopt/sigoptcompute/adapter.py:251]: the tag makes sense here for debugging

File: src/python/zigopt/suggestion/sampler/random.py [hunk 3]
```diff
@@ -45,19 +40,9 @@ def generate_random_suggestions(self, count):
...
-    domain = CategoricalDomain(**asdict(domain_info))
-    if domain_info.priors and not domain_info.constraint_list:
-      samples = domain.generate_random_points_according_to_priors(count)
-    else:
-      samples = domain.generate_quasi_random_points_in_domain(count)
-    # pylint: disable=protected-access
-    suggestion_datas = SCAdapter._make_suggestion_datas(self.experiment, samples)
-    # pylint: enable=protected-access
+    suggestion_datas = self.services.sc_adapter.random_search_next_points(self.experiment, num_to_suggest=count)
+    # NOTE: We could remove this assert later if we decide to make random search non memory-less (i.e., deduping)
```
Comment (malkomes) [inline src/python/zigopt/suggestion/sampler/random.py:45]: nice

**Comment by tjs-intel** [comment]
@harvey-intel you need to run `pip install pipenv && pipenv install && git add Pipfile.lock && git commit -m "Pipfile.lock"`

**Pull request merged**